### PR TITLE
Github workflow for updating dependencies

### PR DESCRIPTION
**Pull request by jdavcs**
## What did you do? 
This is a GitHub workflow that updates all dependencies and submits a PR. It is set to run weekly.

## Why did you make this change?
This should be an improvement to the current way of handling dependencies updates: updating all dependencies more frequently reduces the number of updated items in each iteration. If something breaks, fixing it will be easier.

## How to test the changes? 
n/a

### CODE REVIEW

File: .github/workflows/dependencies.yaml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+on:
+  schedule:
+    - cron: '0 3 * * 6'  # Run every saturday at 3 am.
+jobs:
+  update_dependencies:
+    name: Update dependencies
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: ['3.7']
```
Comment (nsoranzo) [inline .github/workflows/dependencies.yaml]: I'd use our oldest supported version.
```suggestion
        python-version: ['3.6']
```

File: .github/workflows/dependencies.yaml [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+          path: ~/.cache/pip
+          key: pip-cache-${{ matrix.python-version }}-${{ hashFiles('requirements.txt') }}
+      - name: Update dependencies
+        run: | 
+          python -m pip install --upgrade pip setuptools poetry
+          python -m pip install -r lib/galaxy/dependencies/lint-requirements.txt
+          python -m pip freeze -l > lib/galaxy/dependencies/pinned-lint-requirements.txt
+          poetry update -v --lock
+          poetry export -f requirements.txt --without-hashes --output lib/galaxy/dependencies/pinned-requirements.txt
+          poetry export --dev -f requirements.txt --without-hashes --output lib/galaxy/dependencies/dev-requirements.txt
```
Comment (nsoranzo) [inline .github/workflows/dependencies.yaml]: Can't you just re-use the make target?
```suggestion
          python -m venv .venv
          make update-dependencies
```

File: .github/workflows/dependencies.yaml [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+      - name: Create pull request
+        uses: peter-evans/create-pull-request@v3
+        with:
+          commit-message: Update dependencies
+          title: Update dependencies
+          body: Run make update-dependencies
```
Comment (nsoranzo) [inline .github/workflows/dependencies.yaml]:
```suggestion
          commit-message: Update Python dependencies
          title: Update Python dependencies
          body: using `make update-dependencies`.
```

Comment (jdavcs) [reply .github/workflows/dependencies.yaml]: I'd rather not split a sentence across title and body. I think the original body describes precisely what was done.

File: .github/workflows/dependencies.yaml (hunk 2 above)
Comment (jdavcs) [reply .github/workflows/dependencies.yaml]: There was an issue with that. I'll try again - maybe it's as simple as creating a venv.

Comment (jdavcs) [reply .github/workflows/dependencies.yaml]: `./lib/galaxy/dependencies/update_lint_requirements.sh: 7: virtualenv: not found`
@nsoranzo  That script calls `virtualenv`. Do you think we can just use the venv module instead?

Comment (nsoranzo) [reply .github/workflows/dependencies.yaml]: Yes, I think you can replace `virtualenv -p python3` with `python3 -m venv` in `lib/galaxy/dependencies/update_lint_requirements.sh`

File: .github/workflows/dependencies.yaml (hunk 3 above)
Comment (nsoranzo) [reply .github/workflows/dependencies.yaml]: Fair enough, maybe just add the backticks?

File: .github/workflows/dependencies.yaml (hunk 2 above)
Comment (jdavcs) [reply .github/workflows/dependencies.yaml]: ~~I think it's just `python` (not `python3`) - we don't have Python 2 there.~~  EDIT: I'm wrong: it's the script, not the yaml file.

File: .github/workflows/dependencies.yaml [hunk 4]
```diff
@@ -0,0 +1,34 @@
+name: Update dependencies
+on:
+  schedule:
+    - cron: '0 3 * * 6'  # Run every saturday at 3 am.
+jobs:
+  update_dependencies:
+    name: Update dependencies
```
Comment (nsoranzo) [inline .github/workflows/dependencies.yaml:7]: We shouldn't run enable this on forks.
```suggestion
    name: Update dependencies
    if: github.repository_owner == 'galaxyproject'
```

**Review by nsoranzo — APPROVED** [review]

**Pull request merged**